packages/nuxt/src/runtime/plugins/formkit-validation.server.ts

```diff
@@ -0,0 +1,43 @@
+import { defineNuxtPlugin, useRequestEvent } from '#imports'
+import type { FormKitNode } from '@formkit/core'
+import { stopWatch, watchRegistry } from '@formkit/core'
+import { getValidationMessages } from '@formkit/validation'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const context = useRequestEvent()?.context?.formkit
+  if (!context || !context._id) return
+
+  let node: FormKitNode
+
+  // This will be called when the node is initialised
+  const unsub = watchRegistry(context._id, (event) => {
+    if (event.payload) {
+      node = event.payload
+      node.input(context._data)
+    }
+  })
+
+  async function setValidationState() {
+    if (!context) return
+
+    await node.ledger.settled('validating')
```

Comment:
We probably still want this, even if changing `node.input()` to be synchronous — this will ensure async validation rules have had a chance to run as well.